### PR TITLE
Cleanup, document, natspec for contract

### PR DESCRIPTION
**Pull request by 0xAlcibiades**

### CODE REVIEW

File: src/OptionSettlementEngine.sol [hunk 1]
```diff
@@ -411,65 +410,68 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
...
+        // Setup pointers to the option and info.
+        OptionTypeState storage optionTypeState = optionTypeStates[optionKey];
+        Option memory optionRecord = optionTypeState.option;
 
+        // Can't redeem until after expiry
         if (optionRecord.expiryTimestamp > block.timestamp) {
             revert ClaimTooSoon(claimId, optionRecord.expiryTimestamp);
         }
 
-        (uint256 exerciseAmountRedeemed, uint256 underlyingAmountRedeemed) =
```
Comment (0xAlcibiades) [inline src/OptionSettlementEngine.sol:455]: @Flip-Liquid we seem to have been making duplicate calls here.

File: src/OptionSettlementEngine.sol [hunk 2]
```diff
@@ -202,9 +225,13 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
     }
 
     /*//////////////////////////////////////////////////////////////
-    //  Write Options
+    //  Public/External Mutators
     //////////////////////////////////////////////////////////////*/
 
+    //
+    //  Write Options
```
Comment (neodaoist) [inline src/OptionSettlementEngine.sol:232]: On the fence on this subheader pattern fwiw

File: src/OptionSettlementEngine.sol [hunk 3]
```diff
@@ -570,39 +595,66 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
...
-    //////////////////////////////////////////////////////////////*/
-
     /**
-     * @notice Check to see if an option type is already initialized.
+     * @notice Checks to see if an option type is already initialized.
      * @param optionKey The option key to check.
      * @return initialized Whether or not the option type is initialized.
      */
-    function isOptionInitialized(uint160 optionKey) internal view returns (bool initialized) {
+    function isOptionInitialized(uint160 optionKey) private view returns (bool initialized) {
```
Comment (neodaoist) [inline src/OptionSettlementEngine.sol:613]: Prefix with underscore

File: src/OptionSettlementEngine.sol [hunk 4]
```diff
@@ -570,39 +595,66 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
...
     }
 
-    function isClaimInitialized(uint160 optionKey, uint96 claimKey) internal view returns (bool initialized) {
+    /**
+     * @notice Checks to see if an claim is already initialized.
+     * @param optionKey The option key to check.
+     * @param claimKey The claim key to check.
+     * @return initialized Whether or not the claim is initialized.
+     */
+    function isClaimInitialized(uint160 optionKey, uint96 claimKey) private view returns (bool initialized) {
```
Comment (neodaoist) [inline src/OptionSettlementEngine.sol:623]: Prefix with underscore

File: src/interfaces/IOptionSettlementEngine.sol [hunk 5]
```diff
@@ -367,11 +379,13 @@ interface IOptionSettlementEngine {
     /**
      * @notice Gets information about the ERC20 token positions represented by a tokenId.
      * @param tokenId The token id for which to retrieve the Underlying position.
-     * @return underlyingPositions The Underlying struct for the supplied tokenId.
+     * @return underlyingPosition The Underlying struct for the supplied tokenId.
```
Comment (neodaoist) [inline src/interfaces/IOptionSettlementEngine.sol:382]: I think `underlyingPositions` might be more understandable although still not very precise, bc we're already overloading "underlying" as a word, and making it plural might help convey it includes both the underlying asset and the exercise asset.

In fact, renaming `underlying` to `position` in order to de-overload "underlying" word might be better.

**Comment by neodaoist** [comment]
Additional — personally I would remove subheaders in Events, doesn't add that much imo.

If keeping them, need to move `OptionsWritten` to top section

Subheaders in Errors adds more value, but needs more categories, they're wrong/incomplete now.

File: src/OptionSettlementEngine.sol [hunk 6]
```diff
@@ -44,7 +47,10 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
     //  State variables - Internal
     //////////////////////////////////////////////////////////////*/
 
-    /// @notice Accessor for Option contract details
+    /**
+     * @notice Store details about the option, bucket, claim details per option
```
[the remark this answers is not in the document]
Comment (0xAlcibiades) [reply src/OptionSettlementEngine.sol:51]: Me

**Pull request merged**